alicloud/resource_alicloud_drds_instance.go

```diff
@@ -0,0 +1,134 @@
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"specification": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"pay_type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAllowedStringValue([]string{string(Postpaid), string(Prepaid)}),
+			},
+			"vswitch_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"instance_series": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAliCloudDRDSInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+
+	req := drds.CreateCreateDrdsInstanceRequest()
+	req.Description = d.Get("description").(string)
+	req.Type = "1"
+	req.ZoneId = d.Get("zone_id").(string)
+	req.Specification = d.Get("specification").(string)
+	req.PayType = d.Get("pay_type").(string)
+	req.VswitchId = d.Get("vswitch_id").(string)
+	req.InstanceSeries = d.Get("instance_series").(string)
+	req.Quantity = "1"
+	response, err := drdsService.CreateDrdsInstance(req)
+	idList := response.Data.DrdsInstanceIdList.DrdsInstanceId
+	if err != nil || len(idList) != 1 {
+		return fmt.Errorf("failed to create DRDS instance with error: %s", err)
+	}
+	d.SetId(idList[0])
+	return resourceAliCloudDRDSInstanceUpdate(d, meta)
+
+}
+
+func resourceAliCloudDRDSInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+	update := false
+	req := drds.CreateModifyDrdsInstanceDescriptionRequest()
+	req.DrdsInstanceId = d.Id()
+	if d.HasChange("description") && !d.IsNewResource() {
+		update = true
+		req.Description = d.Get("description").(string)
+	}
+	if update {
+		_, err := drdsService.ModifyDrdsInstanceDescription(req)
+		if err != nil {
+			return fmt.Errorf("failed to update Drds instance with error: %s", err)
+		}
+	}
+	return nil
+}
+
+func resourceAliCloudDRDSInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+
+	res, err := drdsService.DescribeDrdsInstance(d.Id())
+	data := res.Data
+	if err != nil || res == nil || data.DrdsInstanceId == "" {
+		d.SetId("")
+		return nil
+	}
+
+	return nil
+}
+
+func resourceAliCloudDRDSInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	drdsService := DrdsService{client}
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
+		req := drds.CreateDescribeDrdsInstanceRequest()
+		req.DrdsInstanceId = d.Id()
+		res, err := drdsService.DescribeDrdsInstance(d.Id())
+		if err != nil {
+			if NotFoundError(err) {
+				return nil
+			}
+		}
+		if res == nil || res.Data.DrdsInstanceId == "" {
+			return nil
+		}
+		removeReq := drds.CreateRemoveDrdsInstanceRequest()
```

Comment:
I think line 122-124 is redundant and the method `DescribeDrdsInstance ` has done it.